### PR TITLE
Split Hive transactional tests to separate job

### PR DESCRIPTION
**Pull request by MiguelWeezardo**

### CODE REVIEW

File: presto-product-tests/bin/product-tests-suite-8-non-generic.sh [hunk 1]
```diff
@@ -14,7 +14,7 @@ suite_exit_code=0
 
 TESTS_HIVE_VERSION_MAJOR="3" TESTS_HIVE_VERSION_MINOR="1" presto-product-tests-launcher/bin/run-launcher test run \
    --environment singlenode-hdp3 \
-    -- -g hdp3_only,storage_formats,hive_transactional \
+    -- -g storage_formats -x hdp3_only,hive_transactional \
```
Comment (findepi) [inline presto-product-tests/bin/product-tests-suite-8-non-generic.sh:17]: hmm ... maybe we just remove `storage_formats` from here?

Comment (losipiuk) [inline presto-product-tests/bin/product-tests-suite-8-non-generic.sh:17]: why do we need -x here?

Comment (MiguelWeezardo) [reply presto-product-tests/bin/product-tests-suite-8-non-generic.sh:17]: I saw that some tests are assigned to both `storage_formats` and `hive_transactional` test groups. With `-g storage_formats` in one job and `-g hdp3_only,hive_transactional` in another wouldn't they be executed in both?

Comment (MiguelWeezardo) [reply presto-product-tests/bin/product-tests-suite-8-non-generic.sh:17]: I saw that some tests are assigned to both `storage_formats` and `hive_transactional` test groups. With `-g storage_formats` in one job and `-g hdp3_only,hive_transactional` in another wouldn't they be executed in both?

Comment (losipiuk) [reply presto-product-tests/bin/product-tests-suite-8-non-generic.sh:17]: It seem there are no tests in both `storage_formats` and ` hdp3_only`. So we can drop `hdp3_only` at least from`-x` arg.
(Even if there was such we should use `profile_specific_tests` for filtering out, not `hdp3_only`)

**Comment by MiguelWeezardo** [comment]
AC

**Comment by findepi** [comment]
@MiguelWeezardo @losipiuk wdyt about https://github.com/prestosql/presto/pull/4222#discussion_r445549276?

ie what i am proposing there is:
```
diff --git presto-product-tests/bin/product-tests-suite-8-non-generic.sh presto-product-tests/bin/product-tests-suite-8-non-generic.sh
index a8af7e1b6e..c81efce553 100755
--- presto-product-tests/bin/product-tests-suite-8-non-generic.sh
+++ presto-product-tests/bin/product-tests-suite-8-non-generic.sh
@@ -14,7 +14,7 @@ suite_exit_code=0

 TESTS_HIVE_VERSION_MAJOR="3" TESTS_HIVE_VERSION_MINOR="1" presto-product-tests-launcher/bin/run-launcher test run \
    --environment singlenode-hdp3 \
-    -- -g hdp3_only,storage_formats,hive_transactional \
+    -- -g hdp3_only,hive_transactional \
     || suite_exit_code=1
```

we run all product tests on hdp3 (via `config-hdp3`).
the way `singlenode-hdp3` env is different from `config-hdp3` currently is:
https://github.com/prestosql/presto/blob/53cdff53d4eb5d62dabe2322ac29b77ad5120729/presto-product-tests-launcher/src/main/resources/docker/presto-product-tests/conf/environment/singlenode-hdp3/hive-site-overrides.xml#L3-L6

**Comment by MiguelWeezardo** [comment]
> @MiguelWeezardo @losipiuk wdyt about [#4222 (comment)](https://github.com/prestosql/presto/pull/4222#discussion_r445549276)?
> 
> ie what i am proposing there is:
> 
> ```
> diff --git presto-product-tests/bin/product-tests-suite-8-non-generic.sh presto-product-tests/bin/product-tests-suite-8-non-generic.sh
> index a8af7e1b6e..c81efce553 100755
> --- presto-product-tests/bin/product-tests-suite-8-non-generic.sh
> +++ presto-product-tests/bin/product-tests-suite-8-non-generic.sh
> @@ -14,7 +14,7 @@ suite_exit_code=0
> 
>  TESTS_HIVE_VERSION_MAJOR="3" TESTS_HIVE_VERSION_MINOR="1" presto-product-tests-launcher/bin/run-launcher test run \
>     --environment singlenode-hdp3 \
> -    -- -g hdp3_only,storage_formats,hive_transactional \
> +    -- -g hdp3_only,hive_transactional \
>      || suite_exit_code=1
> ```
> 
> (without adding a new suite)

This change would decrease test coverage - we'd stop running `storage_formats` in an environment with HDP3 ACID metastore enabled. Suites 1-5 are also running `storage_formats` tests on HDP3, but it doesn't have an ACID metastore.

Is this is an acceptable trade-off? 

> we run all product tests on hdp3 (via `config-hdp3`).
> the way `singlenode-hdp3` env is different from `config-hdp3` currently is:
> 
> https://github.com/prestosql/presto/blob/53cdff53d4eb5d62dabe2322ac29b77ad5120729/presto-product-tests-launcher/src/main/resources/docker/presto-product-tests/conf/environment/singlenode-hdp3/hive-site-overrides.xml#L3-L6

Do we want to run tests on HDP3 without ACID? 
How about enabling ACID by default on all `config-hdp3` environments?

**Comment by findepi** [comment]
> This change would decrease test coverage - we'd stop running `storage_formats` in an environment with HDP3 ACID metastore enabled. Suites 1-5 are also running `storage_formats` tests on HDP3, but it doesn't have an ACID metastore.
> 
> Is this is an acceptable trade-off?

I am proposing that, because I think so. 
But, if we have concerns, maybe we single out those others tests we want to keep and include them instead of the whole group.

> How about enabling ACID by default on all config-hdp3 environments?

see "notes" in https://github.com/prestosql/presto/issues/1841

**Comment by MiguelWeezardo** [comment]
> > This change would decrease test coverage - we'd stop running `storage_formats` in an environment with HDP3 ACID metastore enabled. Suites 1-5 are also running `storage_formats` tests on HDP3, but it doesn't have an ACID metastore.
> > Is this is an acceptable trade-off?
> 
> I am proposing that, because I think so.
> But, if we have concerns, maybe we single out those others tests we want to keep and include them instead of the whole group.

I'm OK with this change, I simply didn't know how much value HDP3 ACID coverage with `storage_formats` tests brings to the table.

> > How about enabling ACID by default on all config-hdp3 environments?
> 
> see "notes" in #1841

I didn't know enabling ACID on HDP3 by default required a lot of work in Presto. 
In that case let's leave the default as it is.

**Comment by findepi** [comment]
@losipiuk @wendigo please take a look

**Comment by losipiuk** [comment]
I was also thinking on enabling acid tables in HIve by default, but given #1841 current solution LGTM

**Comment by findepi** [comment]
Merged, thanks!

**Pull request merged**